doc/news/changes/minor/20211102MaximilianBergbauer

```diff
@@ -0,0 +1,6 @@
+Added: FEEvaluation now supports a user provided shape info object (passed to
+the `reinit` function) with dim univariate shape info objects with different 1D data.
+This enables the use of anisotropic finite elements or quadrature rules which have
+tensor product structure.
+<br>
+(Maximilian Bergbauer, 2021/11/02)
```

Comment:
I suggest we postpone the changelog for now, irrespective of what we choose to do with the `FEEvaluation` interface. We do not really support anisotropic polynomial spaces yet, and one might not want to address them via the additional `shape_info` object.

Thinking about the discussion on `FEEvaluation`: How about skipping that part for now, and instead create a class in user code that derives from `FEEvaluation` and only overrides `reinit` with the specific argument, and then overwrite the `data` pointer in `reinit` with the user-provided one? That way, we keep this functionality out until we see more clearly what functionality we provide for the release, but you can start using all interfaces as usual?

Reply:
Fine with me! We should in parallel start thinking about how to tackle these cases in general.

I will update this PR accordingly.